src/pomerol/DensityMatrix.cpp

```diff
@@ -94,5 +94,47 @@ RealType DensityMatrix::getAverageDoubleOccupancy(ParticleIndex i, ParticleIndex
     NN += (*iter)->getAverageDoubleOccupancy(i,j);
     return NN;
 };
- 
+
+void DensityMatrix::truncateBlocks(RealType Tolerance, bool verbose)
```

Comment:
Junya, do you think this could be in DensityMatrixPart? Essentially you check of there's at least one state with meaningful weight and this could well be done in DensityMatrixPart, where the BlockNumber is fixed.

Reply:
Yes, I moved that into DensityMatrixPart, and it indeed looks much simpler. See 22b64b5. Now, each part has bool whether it was truncated or not.

Additionally, I made a minor change in the verbose output. Previously, the total sum of weights for the retained blocks was printed to check how much weights are lost by truncation. I omitted that in this update.